webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx

```diff
@@ -265,3 +265,16 @@ export default class ChannelSelectorModal extends React.PureComponent<Props, Sta
         );
     }
 }
+
+function compareChannels(a: Channel, b: Channel) {
```

Comment:
I think we have a bunch of different places where we compare channels in similar ways, but they often work slightly differently, so that's why we don't share the code for them. For example, the quick switch modal uses the "last viewed at" time, and channel categories code uses whether or not the channel is muted in its sorting.

As for using the locale, that's probably a mistake here, but 99% of the time, I think sorting is the same between different locales. It's just for things like certain letters or accents that it behaves differently sometimes

Reply:
I like to use the local to be consistent, but 0/5.

Regarding the categories, it removes the muted, but then orders alphabetically (and the muted ones are also ordered alphabetically) so... it is using this at the end of the day.

Probably not the scope of this PR, but that is something we should unify also. We shouldn't have 4 different ways to order channels alphabetically.